openfisca_uk/variables/finance/benefit/family/universal_credit.py

```diff
@@ -1,134 +1,359 @@
+
+    def formula(benunit, period):
+        is_single = benunit("is_single", period)
+        one_over_25 = benunit("eldest_adult_age", period.this_year) >= 25
+        return select(
+            [
+                is_single * not_(one_over_25),
+                is_single * one_over_25,
+                not_(is_single) * not_(one_over_25),
+                not_(is_single) * one_over_25,
+            ],
+            [
+                UCClaimantType.SINGLE_YOUNG,
+                UCClaimantType.SINGLE_OLD,
+                UCClaimantType.COUPLE_YOUNG,
+                UCClaimantType.COUPLE_OLD,
+            ],
+        )
+
+
+class UC_standard_allowance(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"UC Standard Allowance"
+    definition_period = YEAR
+
+    def formula(benunit, period, parameters):
+        SA = parameters(period).benefit.universal_credit.standard_allowance
+        basic_amount = (
+            SA.amount[benunit("UC_claimant_type", period)] * MONTHS_IN_YEAR
+        )
+        return basic_amount
+
+
+class is_child_born_before_child_limit(Variable):
     value_type = bool
+    entity = Person
+    label = u"Born before child limit (exempt)"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        UC = parameters(period).benefit.universal_credit
+        start_year = UC.elements.child.limit.start_year
+        born_before_limit = person("birth_year", period) < start_year
+        return person("is_child", period) * born_before_limit
+
+
+class UC_individual_child_element(Variable):
+    value_type = float
+    entity = Person
+    label = u"UC child element for the child"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        child_index = person("child_index", period)
+        born_before_limit = person("is_child_born_before_child_limit", period)
+        is_eligible = child_index <= person.benunit(
+            "num_UC_eligible_children", period
+        )
+        UC = parameters(period).benefit.universal_credit
+        return (
+            select(
+                [
+                    (child_index == 1) & born_before_limit,
+                    child_index == 1,
+                    (child_index > 1) & is_eligible,
+                    True,
+                ],
+                [
+                    UC.elements.child.first.higher_amount,
+                    UC.elements.child.amount,
+                    UC.elements.child.amount,
+                    0,
+                ],
+            )
+            * MONTHS_IN_YEAR
+        )
+
+
+class num_UC_eligible_children(Variable):
+    value_type = int
     entity = BenUnit
-    label = u"Whether this family is imputed to claim UC"
+    label = u"Eligible children"
     definition_period = YEAR
 
     def formula(benunit, period, parameters):
-        already_claiming = (
-            aggr(benunit, period, ["universal_credit_reported"]) > 0
+        born_before_limit = benunit.sum(
```

Comment:
currently reads like a bool
```suggestion
        children_born_before_limit = benunit.sum(
```

Reply:
Done